### PR TITLE
RSDK-6406 - Change implementation of MoveOnMap

### PR DESCRIPTION
**Pull request by nfranczak**
We allow users to specify static obstacles which are meant to be avoided at plan time.

### CODE REVIEW

**Review by biotinker — APPROVED** [review]
Love that this was so easy to add

File: services/motion/builtin/move_request.go [hunk 1]
```diff
@@ -612,14 +612,16 @@ func (ms *builtIn) newMoveOnMapRequest(
 		return nil, err
 	}
 
+	req.Obstacles = append(req.Obstacles, octree)
```
Comment (raybjork) [inline services/motion/builtin/move_request.go:615]: Is it true that all these obstacles are in the same frame?  Do they need to be?

File: services/motion/builtin/builtin_test.go [hunk 2]
```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
...
+
+		// construct move request
+		planExecutor, err := ms.(*builtIn).newMoveOnMapRequest(ctx, req, nil, 0)
+		test.That(t, err, test.ShouldBeNil)
+		mr, ok := planExecutor.(*moveRequest)
+		test.That(t, ok, test.ShouldBeTrue)
+
+		// construct plan
+		planResp, err := mr.Plan(ctx)
+		test.That(t, err, test.ShouldBeError, errors.New("context deadline exceeded"))
```
Comment (raybjork) [inline services/motion/builtin/builtin_test.go:1563]: How long does this take to run? I don't see a timeout on the context

File: services/motion/builtin/builtin_test.go [hunk 3]
```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
...
+		test.That(t, err, test.ShouldBeNil)
+
+		obstacleRight, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(r3.Vector{0.82183e3, 0.64589e3, 0}),
+			r3.Vector{10, 400, 10},
+			"obstacleRight",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		req.Obstacles = []spatialmath.Geometry{obstacleTop, obstacleBottom, obstacleLeft, obstacleRight}
```
Comment (raybjork) [inline services/motion/builtin/builtin_test.go:1553]: Why use multiple obstacles here?  One at the goal location should be sufficient I think

File: services/motion/builtin/move_request.go (hunk 1 above)
Comment (nfranczak) [reply services/motion/builtin/move_request.go:615]: the api message gives us proto geometries which have on way of telling us with what frame they are defined with respect to. It's on the user to know that at call time.

File: services/motion/builtin/builtin_test.go (hunk 2 above)
Comment (nfranczak) [reply services/motion/builtin/builtin_test.go:1563]:
```
	extra := map[string]interface{}{
		"motion_profile": "position_only",
		"timeout":        15.,
		"smooth_iter":    10.,
	}
```
we set a timeout through extras

File: services/motion/builtin/builtin_test.go [hunk 4]
```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
...
+			[]*referenceframe.GeometriesInFrame{
+				referenceframe.NewGeometriesInFrame(
+					referenceframe.World,
+					[]spatialmath.Geometry{obstacleRight},
+				),
+			}, nil,
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = motionplan.CheckPlan(
```
Comment (raybjork) [inline services/motion/builtin/builtin_test.go:1506]: Why are we checking this?

**Review by raybjork — CHANGES_REQUESTED** [review]

File: services/motion/builtin/builtin_test.go (hunk 3 above)
Comment (biotinker) [reply services/motion/builtin/builtin_test.go:1553]: I specifically requested this :)

Having an obstacle overlap the goal location yields different failure modes than having the base be able to validly exist at the goal location, but be unable to connect to the start.

File: services/motion/builtin/builtin_test.go (hunk 2 above)
Comment (raybjork) [reply services/motion/builtin/builtin_test.go:1563]: 15 seems kind of high, can we get away with something lower?

Comment (raybjork) [reply services/motion/builtin/builtin_test.go:1563]: Is 5 too aggressive?

Comment (nfranczak) [reply services/motion/builtin/builtin_test.go:1563]: i think 5 would work. the success case is a solving for path for a trivial goal - which takes `0.4` seconds on my laptop.

**Review by raybjork — APPROVED** [review]
The actual code change here is hilarious to me.  Nice job

File: services/motion/builtin/builtin_test.go [hunk 5]
```diff
@@ -1400,6 +1401,170 @@ func TestMoveOnGlobe(t *testing.T) {
...
+
+	req := motion.MoveOnMapReq{
+		ComponentName: injectBase.Name(),
+		Destination:   goal,
+		SlamName:      injectSlam.Name(),
+		MotionCfg:     &motion.MotionConfiguration{PlanDeviationMM: 0.01},
+		Extra:         extra,
+	}
+
+	t.Run("one obstacle", func(t *testing.T) {
```
Comment (nfranczak) [inline services/motion/builtin/builtin_test.go:1462]: Adding context for the following test case
the robot would be in the following octagon
og

I would then define a static obstacle which would generate a path which curves to the right


i would then use the CheckPlan function to show that with the following new set of obstacles the plan fails proving that the path went to the right and avoided the obstacle on the left which we defined at run time
wts fail

**Pull request merged**